File: PS6/sentimental-readability/readability.py
from cs50 import get_string

sentence = get_string("Text: ")
letter_count = 0
word_count = 1
sentence_count = 0

for char in range(len(sentence)):
    if sentence[char].isalnum():
        letter_count += 1
    elif sentence[char] == ' ' and letter_count > 0:
        word_count += 1
    elif sentence[char] in ['.', '?', '!']:
        sentence_count += 1
    else:
        continue

L = (letter_count / word_count) * 100
S = (sentence_count / word_count) * 100

index = 0.0588 * L - 0.296 * S - 15.8

if 0 < index < 16:
    print(f"Grade {round(index)}")
elif index < 1:
    print(f"Before Grade 1")
else:
    print("Grade 16+")

File: PS2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: "); // input
    int slenght = strlen(text);         // convert lenght of string into an integer

    for (int n = 0; n < slenght; n++)
    {
        text[n] = toupper(text[n]);
    }

    int letters = 0;
    int words = 0;
    int sentences = 0;

    for (int n = 0; n < slenght; n++)
    {
        if ('A' <= text[n] && text[n] <= 'Z')
            letters++;
        if ((text[n] == ' ' || text[n] == '.' || text[n] == '?' || text[n] == '!' || text[n] == ':' || text[n] == ',') &&
            'A' <= text[n - 1] && text[n - 1] <= 'Z')
            words++;
        if ('A' <= text[n - 1] && text[n - 1] <= 'Z' && (text[n] == '.' || text[n] == '?' || text[n] == '!'))
            sentences++;
    }

    float L = ((float) letters / words) * 100;
    float S = ((float) sentences / words) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
        printf("Before Grade 1\n");
    else if (16 < index)
        printf("Grade 16+\n");
    else
        printf("Grade %i\n", index);
}
